modules/vba_documents/spec/sidekiq/slack_status_notifier_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require './modules/vba_documents/app/sidekiq/vba_documents/slack_status_notifier'
+
+RSpec.describe 'VBADocuments::SlackStatusNotifier', type: :job do
+  let(:slack_messenger) { instance_double(VBADocuments::Slack::Messenger.class_name) }
+  let(:slack_enabled) { true }
+
+  before do
+    allow(VBADocuments::Slack::Messenger).to receive(:new).and_return(slack_messenger)
+    allow(slack_messenger).to receive(:notify!)
+    allow(Settings.vba_documents.slack).to receive(:enabled).and_return(slack_enabled)
+    @job = VBADocuments::SlackStatusNotifier.new
+    @results = nil
+  end
+
+  context 'when flag is disabled' do
+    let(:slack_enabled) { false }
+
+    it 'does nothing' do
+      with_settings(Settings.vba_documents.slack, enabled: false) do
+        @results = @job.perform
+        expect(slack_messenger).not_to have_received(:notify!)
+      end
+    end
+  end
+
+  context 'when no new expired uploads were found' do
+    before do
+      VBADocuments::UploadSubmission.new(status: 'pending', consumer_name: 'sometech').save!
+      VBADocuments::UploadSubmission.new(status: 'uploaded', consumer_name: 'sometech').save!
+      VBADocuments::UploadSubmission.new(status: 'received', consumer_name: 'sometech').save!
+      VBADocuments::UploadSubmission.new(status: 'processing', consumer_name: 'sometech').save!
+      VBADocuments::UploadSubmission.new(status: 'success', consumer_name: 'sometech').save!
+      VBADocuments::UploadSubmission.new(status: 'vbms', consumer_name: 'sometech').save!
+      VBADocuments::UploadSubmission.new(status: 'error', consumer_name: 'sometech').save!
+      VBADocuments::UploadSubmission.new(status: 'expired', consumer_name: 'sometech', created_at: 5.days.ago).save!
```

Comment:
Is there any reason why you didn't use the existing `:upload_submission` factory here? E.g.
`create(:upload_submission, status: 'pending', consumer_name: 'sometech')`

Calling a class directly to create objects in specs isn't really standard for us and I'm surprised that Rubocop didn't catch it.

Reply:
I copied that from the original slacknotifier tests